File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.cpp
class Solution {
public:
    int numTrees(int n) {
        vector<int> dp(n+1);
        dp[0] = dp[1] = 1;
        for(int i = 2; i <= n; i++) {
            int l=0,r=i-1;
            while(l<=i-1){
                dp[i]+=dp[l]*dp[r];
                l++;
                r--;
            }
        }
           
        return dp[n];
    }
};

File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.py
class Solution:
    def numTrees(self, n: int) -> int:
        dp = [0] * (n+1)
        dp[0] = 1
        dp[1] = 1;
        for i in range(2,n+1): 
            l = 0
            r = i-1
            while(l<=i-1):
                dp[i] += (dp[l]*dp[r]);
                l = l+1
                r = r-1
           
        return dp[n];